plugins/package-managers/maven/src/main/kotlin/utils/MavenSupport.kt

```diff
@@ -622,6 +622,37 @@ private fun createContainer(): PlexusContainer {
     }
 }
 
+/** The namespace for the Maven Tycho build extension. */
+private const val TYCHO_NAMESPACE = "org.eclipse.tycho"
+
+/** The ID for the Maven Tycho build extension. */
+private const val TYCHO_ID = "tycho-build"
+
+/** The path to the subfolder containing core extensions for Maven. */
+private const val EXTENSIONS_PATH = ".mvn/extensions.xml"
+
+/**
+ * Return *true* if the given [file] points to a Maven Tycho project. The [file] can either reference the project
+ * folder directly or a file within the project folder.
+ */
+internal fun isTychoProject(file: File): Boolean {
+    val root = file.takeIf { it.isDirectory } ?: file.parentFile
+
+    return root?.resolve(EXTENSIONS_PATH)?.takeIf { it.isFile }?.let { extFile ->
+        val content = extFile.readText()
+        TYCHO_NAMESPACE in content && TYCHO_ID in content
+    } == true
+}
+
+/**
+ * Filter out all files from the given [definitionFiles] that belong to sub folders of one of the given [roots],
+ * including the [roots] themselves. This function can be used to detect whole multi-module projects.
+ */
+internal fun filterChildFolders(roots: Collection<File>, definitionFiles: Collection<File>): List<File> =
+    definitionFiles.filterNot { file ->
+        roots.any { file.startsWith(it) }
+    }
```

Comment:
IMO this does not deserve a separate function. The code is rather trivial and could even be a one-liner. Plus, the function will one be used once.

So I'd prefer to inline this to where it's actually being used, instead of adding an unused function beforehand just to add a test for it.

The test (which is longer than the function itself) I'd drop completely, esp. as you later add another test for `mapDefinitionFiles()` which basically tests the same thing.

Reply:
The function has been removed.